src/redis_lock/__init__.py

```diff
@@ -88,18 +92,21 @@ def id(self):
     def get_owner_id(self):
         return self._client.get(self._name)
 
-    def acquire(self, blocking=True):
+    def acquire(self, blocking=True, timeout=None):
         logger.debug("Getting %r ...", self._name)
 
         if self._held:
             raise AlreadyAcquired("Already aquired from this Lock instance.")
 
         busy = True
+        timeout = self._timeout if timeout is None else int(timeout)
         while busy:
             busy = not self._client.set(self._name, self._id, nx=True, ex=self._expire)
             if busy:
                 if blocking:
-                    self._client.blpop(self._signal, self._expire or 0)
+                    timed_out = not self._client.blpop(self._signal, timeout or self._expire or 0)
```

Comment:
Here we should pick the minimum of [timeout, self._expire or 0].

Ooooor, maybe it's better to raise error if timeout>self._expire (if expire is set). Maybe it's better to not second-guess the user.